samples/tfm_integration/psa_crypto/src/psa_crypto.c

```diff
@@ -91,105 +107,421 @@ static psa_status_t crp_get_pub_key(psa_key_id_t key_id,
-			goto err;
-		}
-
-		/* Display the private key. */
-		LOG_INF("Private key data:");
-		al_dump_log();
-		sf_hex_tabulate_16(&crp_fmt, data_out, data_len);
-
-		/* Close the key to free up the volatile slot. */
-		status = al_psa_status(
-			psa_close_key(key_handle),
-			__func__);
-		if (status != PSA_SUCCESS) {
-			LOG_ERR("Failed to close persistent key.");
-			goto err;
-		}
-	}
+        /* Close the key to free up the volatile slot. */
+        status = al_psa_status(
+            psa_close_key(key_handle),
+            __func__);
+        if (status != PSA_SUCCESS) {
+            LOG_ERR("Failed to close persistent key.");
+            goto err;
+        }
+    }
 
-	return status;
+    return status;
 err:
 	al_dump_log();
 	return status;
 }
+#endif // CONFIG_PSA_IMPORT_KEY
+
+/**
+ * @brief PSA Random number generator wrapper for Mbed TLS
+ */
+static int psa_rng_for_mbedtls(void *p_rng,
+                               unsigned char *output, size_t output_len)
+{
+    (void)p_rng;
+
+    return psa_generate_random(output, output_len);
+}
+
+/**
+ * @brief Generates device certificate signing request (CSR) using Mbed TLS
+ * X.509 and TF-M crypto service.
+ */
+void crp_generate_csr(void)
+{
+    psa_status_t status;
+    psa_key_id_t key_slot = 1;
+    psa_key_handle_t key_handle;
+
+    unsigned char output_buf[1024];
+	unsigned char json_encoded_buf[1024];
+
+    mbedtls_pk_context pk_key_container;
+    mbedtls_x509write_csr req;
+
+    struct csr_json_struct csr_json = {
+        .CSR = output_buf
+    };
+
+    /* Initialize Mbed TLS structures. */
+    mbedtls_x509write_csr_init( &req );
+    mbedtls_pk_init( &pk_key_container );
+    memset( output_buf, 0, sizeof( output_buf ) );
+
+    /* Initialize crypto API. */
+    LOG_INF("Initialising PSA crypto");
+	al_dump_log();
+
+    status = al_psa_status(psa_crypto_init(), __func__);
+    if (status != PSA_SUCCESS) {
+        LOG_ERR("Crypto init failed.");
+        goto err;
+    }
+
+    LOG_INF("PSA crypto init completed");
+	al_dump_log();
+
+    /* prime256v1 (ecdsa-with-SHA256) private key. */
+#if CONFIG_PSA_IMPORT_KEY
+#if CONFIG_PRIVATE_KEY_STATIC
+    /* This value is based on the private key in user.pem,
+     * which can be viewed viw the following command:
+     *
+     *   $ openssl ec -in user.pem -text -noout
+     */
+    uint8_t priv_key_data[32] = {
+        0x14, 0xbc, 0xb9, 0x53, 0xa4, 0xee, 0xed, 0x50,
+        0x09, 0x36, 0x92, 0x07, 0x1d, 0xdb, 0x24, 0x2c,
+        0xef, 0xf9, 0x57, 0x92, 0x40, 0x4f, 0x49, 0xaa,
+        0xd0, 0x7c, 0x5b, 0x3f, 0x26, 0xa7, 0x80, 0x48
+    };
+#else // CONFIG_PRIVATE_KEY_STATIC
+    /* Randomly generate the private key. */
+    uint8_t priv_key_data[32] = { 0 };
+
+    LOG_INF("Generate rnadom data for private key");
+	al_dump_log();
+
+    psa_generate_random(priv_key_data, sizeof(priv_key_data));
+    LOG_INF("Random data generation for private key completed");
+	al_dump_log();
+
+#endif // CONFIG_PRIVATE_KEY_STATIC
```

Comment:
Multiple usage of `CONFIG_PRiVATE_KEY_STATIC`

Reply:
I've updated the comments next to the `if else` blocks. Hope it's easier to understand now.